src/game.cpp

```diff
@@ -2722,19 +2722,43 @@ void Game::playerAcceptTrade(uint32_t playerId)
 		}
 
 		bool isSuccess = false;
+		
+		// assign item as itemType
+		const ItemType& partnerItemType = Item::items[partnerTradeItem->getID()];
+		const ItemType& playerItemType = Item::items[playerTradeItem->getID()];
+
+		// retrieve slot type for given item
+		slots_t partnerSlotItem = getSlotType(partnerItemType);
```

Comment:
```
slots_t partnerSlotItem = getSlotType(Item::items[partnerTradeItem->getID()]);
slots_t playerSlotItem = getSlotType(Item::items[playerTradeItem->getID()]);
```
and remove these variables for itemtypes

Reply:
This is what I was looking for! Didn't know that was possible. Thanks. Anyway, the code has been refactored.